common/etc/nginx/include/s3gateway.js

```diff
@@ -477,234 +442,7 @@ function _requireEnvVars(envVarName) {
-            return;
-        }
-    }
-    try {
-        awscred.writeCredentials(r, credentials);
-    } catch (e) {
-        utils.debug_log(r, `Could not write credentials: ${e}`);
-        r.return(500);
-        return;
-    }
-    r.return(200);
-}
-
-/**
- * Get the credentials needed to generate AWS signatures from the ECS
- * (Elastic Container Service) metadata endpoint.
- *
- * @param credentialsUri {string} endpoint to get credentials from
- * @returns {Promise<{accessKeyId: (string), secretAccessKey: (string), sessionToken: (string), expiration: (string)}>}
- * @private
- */
-async function _fetchEcsRoleCredentials(credentialsUri) {
-    const resp = await ngx.fetch(credentialsUri);
-    if (!resp.ok) {
-        throw 'Credentials endpoint response was not ok.';
-    }
-    const creds = await resp.json();
-
-    return {
-        accessKeyId: creds.AccessKeyId,
-        secretAccessKey: creds.SecretAccessKey,
-        sessionToken: creds.Token,
-        expiration: creds.Expiration,
-    };
-}
-
-/**
- * Get the credentials needed to generate AWS signatures from the EC2
- * metadata endpoint.
- *
- * @returns {Promise<{accessKeyId: (string), secretAccessKey: (string), sessionToken: (string), expiration: (string)}>}
- * @private
- */
-async function _fetchEC2RoleCredentials() {
-    const tokenResp = await ngx.fetch(EC2_IMDS_TOKEN_ENDPOINT, {
-        headers: {
-            'x-aws-ec2-metadata-token-ttl-seconds': '21600',
-        },
-        method: 'PUT',
-    });
-    const token = await tokenResp.text();
-    let resp = await ngx.fetch(EC2_IMDS_SECURITY_CREDENTIALS_ENDPOINT, {
-        headers: {
-            'x-aws-ec2-metadata-token': token,
-        },
-    });
-    /* This _might_ get multiple possible roles in other scenarios, however,
-       EC2 supports attaching one role only.It should therefore be safe to take
-       the whole output, even given IMDS _might_ (?) be able to return multiple
-       roles. */
-    const credName = await resp.text();
-    if (credName === "") {
-        throw 'No credentials available for EC2 instance';
-    }
-    resp = await ngx.fetch(EC2_IMDS_SECURITY_CREDENTIALS_ENDPOINT + credName, {
-        headers: {
-            'x-aws-ec2-metadata-token': token,
-        },
-    });
-    const creds = await resp.json();
-
-    return {
-        accessKeyId: creds.AccessKeyId,
-        secretAccessKey: creds.SecretAccessKey,
-        sessionToken: creds.Token,
-        expiration: creds.Expiration,
-    };
-}
-
-/**
- * Get the credentials by assuming calling AssumeRoleWithWebIdentity with the environment variable
- * values ROLE_ARN, AWS_WEB_IDENTITY_TOKEN_FILE and HOSTNAME
- *
- * @returns {Promise<{accessKeyId: (string), secretAccessKey: (string), sessionToken: (string), expiration: (string)}>}
- * @private
- */
-async function _fetchWebIdentityCredentials(r) {
-    const arn = process.env['AWS_ROLE_ARN'];
-    const name = process.env['HOSTNAME'] || 'nginx-s3-gateway';
```

Comment:
How is the hardcoded value `nginx-s3-gateway` being used? It would be useful to have a comment about it.

Reply:
- This is the existing hardcoded value in this repo's codebase. Looks like it is used as a default Role Session Name (`nginx-s3-gateway`) in case `HOSTNAME` is missed.

  - https://github.com/nginxinc/nginx-s3-gateway/blob/master/docs/getting_started.md?plain=1#L54

- As we are moving this function into the common library(`awscredentials.js`), we can consider one of the following options.
  - option 1. remove this default value (`nginx-s3-gateway`) so customers must define the env var (`HOSTNAME`).
  - option 2. replace this value with other variable (e.g. `$defaultHostName`).
  - option 3. any other suggestion?
  - FYI:
    - https://github.com/nginxinc/nginx-s3-gateway/pull/126/files#r1190281902
    - https://github.com/shawnhankim/nginx-s3-gateway/blob/awssig-env/common/etc/nginx/include/awscredentials.js#L373
    - https://github.com/shawnhankim/nginx-s3-gateway/blob/awssig-env/common/etc/nginx/templates/default.conf.template#L27

I was wondering what would be your suggestion.

Reply:
- I have removed this hardcoded value in this NJS code, because this NJS can be used for not only S3 but also other services such as Lambda.
- I have set this default value in the https://github.com/shawnhankim/nginx-s3-gateway/blob/awssig-env/common/docker-entrypoint.d/00-check-for-required-env.sh#L58 as the following example instead:
  ```bash
   elif [[ -v AWS_WEB_IDENTITY_TOKEN_FILE ]]; then
     echo "Running inside EKS with IAM roles for service accounts"
     if [[ ! -v HOSTNAME ]]; then
       # This environment value is used for Role Session Name. The default value is
       # set as a nginx-s3-gateway unless the value is defined.
       HOSTNAME="nginx-s3-gateway"
     fi
  ```
- Let me know if you have other suggestion, or if you agree with this approach.

Reply:
This approach is I believe is the best we can do. I was considering modifying the startup scripts such that if the environment variable `HOSTNAME` is not set that script would run `hostname` and then set `HOSTNAME` to the value returned.

Reply:
- Thanks for your opinion and original intent.
- Yeah, I was also confused about the [comment and variable](https://github.com/nginxinc/nginx-s3-gateway/blob/master/docs/getting_started.md?plain=1#L54). So I have replaced `HOSTNAME` with `AWS_ROLE_SESSION_NAME` and added the depreciation message of the variable of `HOSTNAME`.